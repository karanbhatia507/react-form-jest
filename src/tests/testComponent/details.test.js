import React from 'react';
import { mount } from 'enzyme';
import Details from '../../components/details'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import Modal from "../../components/details";
import Button from "../../components/details";

configure({ adapter: new Adapter() })

describe('Testing Details Component', () => {
   
    it('Details component is defined', () => {
    const component = mount(<Details />)
    expect(component).toBeDefined()
   })

   it('Contains information about user', () => {
    const component = mount(<Details name="Karan" contact="98995XXXXX"/>)
    const tree = toJson(component)
    expect(component.prop('name')).toContain("Karan")
    expect(typeof component.prop('contact')).toBe("string")
    expect(tree).toMatchSnapshot()
   })

//    it('handleChange works as expected', () => {
//     const component = mount(<Details handleChange={() => {}} />)
//     component.find("input").simulate("change", {
//         target: { value: "hello" }
//     });
//     expect(component.find("input").props().value).toEqual("hello");
//    });

   it('has props greater than 3', () => {
    const component = mount(<Details name="Karan" contact="98995XXXXX" email="abc@gmail.com" enquiry="hey whats up people...?" />)
    expect(component.prop('name').length).toBeGreaterThan(3); 
    expect(component.find(Button).children()).toHaveLength(1) 
    expect(component.find(Modal).children()).toHaveLength(1)
    expect(component.prop('name')).toMatch(/ran/)
    expect(component.prop('name')).not.toMatch(/rak/)
   })
})