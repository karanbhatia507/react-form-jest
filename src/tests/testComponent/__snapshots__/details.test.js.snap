// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Details Component Contains information about user 1`] = `
<Details
  contact="98995XXXXX"
  name="Karan"
>
  <div>
    <Button
      className="submt_btn"
      color="danger"
      onClick={[Function]}
      tag="button"
    >
      <button
        aria-label={null}
        className="submt_btn btn btn-danger"
        onClick={[Function]}
        type="button"
      />
    </Button>
    <Modal
      autoFocus={true}
      backdrop={true}
      backdropTransition={
        Object {
          "mountOnEnter": true,
          "timeout": 150,
        }
      }
      centered={false}
      fade={true}
      isOpen={false}
      keyboard={true}
      modalTransition={
        Object {
          "timeout": 300,
        }
      }
      onClosed={[Function]}
      onOpened={[Function]}
      role="dialog"
      toggle={[Function]}
      zIndex={1050}
    />
  </div>
</Details>
`;
